creator-node/src/services/stateMachineManager/stateReconciliation/updateReplicaSet.jobProcessor.ts

```diff
@@ -87,26 +93,50 @@ const updateReplicaSetJobProcessor = async function ({
         enableIdentity: true,
         logger
       })
+    } catch (e) {
+      error = error || 'failure_init_audius_libs'
+      errorMsg = 'failed to init libs'
+      logger.error(`ERROR ${errorMsg} - ${(e as Error).message}`)
+    }
+
+    try {
       const { services: healthyServicesMap } =
         await audiusLibs.ServiceProvider.autoSelectCreatorNodes({
           performSyncCheck: false,
           whitelist: reconfigNodeWhitelist,
           log: true
         })
       healthyNodes = Object.keys(healthyServicesMap || {})
-      if (healthyNodes.length === 0)
+      if (healthyNodes.length === 0) {
         throw new Error(
           'Auto-selecting Content Nodes returned an empty list of healthy nodes.'
         )
+      }
       await cacheHealthyNodes(healthyNodes)
     } catch (e: any) {
+      error = error || 'failure_find_healthy_nodes'
       const errorMsg = `Error initting libs and auto-selecting creator nodes: ${e.message}: ${e.stack}`
       logger.error(errorMsg)
+
+      // Make metrics to record
+      metricsToRecord = [
+        makeHistogramToRecord(
+          METRIC_NAMES[
+            `STATE_MACHINE_${QUEUE_NAMES.UPDATE_REPLICA_SET}_JOB_DURATION_SECONDS_HISTOGRAM`
+          ],
+          (Date.now() - startTimeMs) / 1000, // Metric is in seconds
+          {
+            result: error || 'success'
+          }
+        )
+      ]
```

Comment:
since you removed this logic from `processJob` i think we need to now add it here?
```
  if (jobName === QUEUE_NAMES.UPDATE_REPLICA_SET) {
    const { issuedReconfig, newReplicaSet } = jobResult
    return {
      issuedReconfig: issuedReconfig || 'false',
      reconfigType: _.snakeCase(newReplicaSet?.reconfigType || 'null')
    }
  }
```

Reply:
better approach would be to only record the metric in one place, similar to `issueSyncRequest.jobProcessor` to avoid this kind of confusion
thoughts?

Reply:
Yup, that does need to be there